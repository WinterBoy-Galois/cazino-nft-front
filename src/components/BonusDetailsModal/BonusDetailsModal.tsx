import React from 'react';
import Modal from '../Modal';
import { useTranslation } from 'react-i18next';
import clsx from 'clsx';
import styles from './BonusDetailsModal.module.scss';
import { formatBitcoin } from '../../common/util/format.util';
import Bitcoin from '../icons/social/Bitcoin';

interface IProps {
  show: boolean;
  onClose?: () => void;
  givenAt: string;
  type: string;
  wager: number;
  amount: number;
}

const BonusDetailsModal: React.FC<IProps> = ({ show, onClose, givenAt, type, wager, amount }) => {
  const { t } = useTranslation(['modals']);

  return (
    <Modal
      show={show}
      onClose={onClose}
      title={t('bonusDetails.title')}
      modalClassName="circle_warning"
    >
      <div className={clsx(styles.section)}>
        <div className={clsx(styles.section__row)}>
          <div className={clsx(styles.section__label)}>{t('bonusDetails.time')}</div>
          <div className={clsx(styles.section__value)}>{givenAt}</div>
        </div>

        <div className={clsx(styles.section__row)}>
          <div className={clsx(styles.section__label)}>{t('bonusDetails.type')}</div>
          <div className={clsx(styles.section__value)}>{type}</div>
        </div>

        <div className={clsx(styles.section__row)}>
          <div className={clsx(styles.section__label)}>{t('bonusDetails.wager')}</div>
          <div className={clsx(styles.section__value, styles.section__value__amount)}>
            <Bitcoin />
            {formatBitcoin(wager)}
          </div>
        </div>

        <div className={clsx(styles.section__row)}>
          <div className={clsx(styles.section__label)}>{t('bonusDetails.amount')}</div>
          <div
            className={clsx(
              styles.section__value,
              styles.section__value__amount,
              styles.section__value__bonus
            )}
          >
            <Bitcoin />+{formatBitcoin(amount)}
          </div>
        </div>
      </div>
    </Modal>
  );
};

export default BonusDetailsModal;
