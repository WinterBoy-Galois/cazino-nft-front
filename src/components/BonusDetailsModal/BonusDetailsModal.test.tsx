import React from 'react';
import { render } from '@testing-library/react';
import BonusDetailsModal from '.';
import { LocationProvider } from '@reach/router';

describe('BonusDetailsModal', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(
      <LocationProvider>
        <BonusDetailsModal
          show={true}
          givenAt="02/16/2021 13:44:11"
          claimedAt="02/18/2021 01:41:36"
          type="DAILY"
          wager={0.00069079}
          amount={0.00001234}
        />
      </LocationProvider>
    );

    // Assert
    expect(container).toMatchSnapshot();
  });
});
