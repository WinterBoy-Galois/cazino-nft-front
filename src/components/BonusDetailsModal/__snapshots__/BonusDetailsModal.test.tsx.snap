// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BonusDetailsModal should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div>
      <div
        class="container"
        style="height: 768px;"
      >
        <div
          aria-modal="true"
          class="circle_warning modal scale--enter scale--enter-active"
          role="dialog"
        >
          <div
            class="modal__header"
          >
            <div
              class="modal__header__headline"
            >
              <svg
                class="icon icon__warning"
                viewBox="0 0 35 35"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  data-name="Layer 2"
                  id="Layer_2"
                >
                  <g
                    data-name="Слой 5"
                    id="Слой_5"
                  >
                    <path
                      class="cls-1"
                      d="M17.5,0A17.5,17.5,0,1,0,35,17.5,17.51,17.51,0,0,0,17.5,0Zm2.56,24.54a2.14,2.14,0,0,1-2.14,2.14h-.83A2.14,2.14,0,0,1,15,24.54V17.69a2.14,2.14,0,0,1,2.14-2.15h.83a2.14,2.14,0,0,1,2.14,2.15ZM17.5,13.3a3.09,3.09,0,1,1,3.1-3.09A3.09,3.09,0,0,1,17.5,13.3Z"
                    />
                  </g>
                </g>
              </svg>
              BONUS DETAILS
            </div>
            <div
              class="modal__header__close"
            >
              <svg
                class="icon icon__close"
                viewBox="0 0 365.696 365.696"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M243.188 182.86L356.32 69.726c12.5-12.5 12.5-32.766 0-45.247L341.238 9.398c-12.504-12.503-32.77-12.503-45.25 0L182.86 122.528 69.727 9.374c-12.5-12.5-32.766-12.5-45.247 0L9.375 24.457c-12.5 12.504-12.5 32.77 0 45.25l113.152 113.152L9.398 295.99c-12.503 12.503-12.503 32.769 0 45.25L24.48 356.32c12.5 12.5 32.766 12.5 45.247 0l113.132-113.132L295.99 356.32c12.503 12.5 32.769 12.5 45.25 0l15.081-15.082c12.5-12.504 12.5-32.77 0-45.25zm0 0"
                />
              </svg>
            </div>
          </div>
          <div
            class="modal__main"
          >
            <div
              class="section"
            >
              <div
                class="section__row"
              >
                <div
                  class="section__label"
                >
                  Date / time
                </div>
                <div
                  class="section__value"
                >
                  02/16/2021 13:44:11
                </div>
              </div>
              <div
                class="section__row"
              >
                <div
                  class="section__label"
                >
                  CLAIM AT
                </div>
                <div
                  class="section__value"
                >
                  02/18/2021 01:41:36
                </div>
              </div>
              <div
                class="section__row"
              >
                <div
                  class="section__label"
                >
                  TYPE
                </div>
                <div
                  class="section__value"
                >
                  DAILY
                </div>
              </div>
              <div
                class="section__row"
              >
                <div
                  class="section__label"
                >
                  WAGER
                </div>
                <div
                  class="section__value section__value__amount"
                >
                  <svg
                    class="fill"
                    viewBox="0 0 64 64"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      transform="translate(0.00630876,-0.00301984)"
                    >
                      <path
                        d="m63.033,39.744c-4.274,17.143-21.637,27.576-38.782,23.301-17.138-4.274-27.571-21.638-23.295-38.78,4.272-17.145,21.635-27.579,38.775-23.305,17.144,4.274,27.576,21.64,23.302,38.784z"
                      />
                      <path
                        class="innerFill"
                        d="m46.103,27.444c0.637-4.258-2.605-6.547-7.038-8.074l1.438-5.768-3.511-0.875-1.4,5.616c-0.923-0.23-1.871-0.447-2.813-0.662l1.41-5.653-3.509-0.875-1.439,5.766c-0.764-0.174-1.514-0.346-2.242-0.527l0.004-0.018-4.842-1.209-0.934,3.75s2.605,0.597,2.55,0.634c1.422,0.355,1.679,1.296,1.636,2.042l-1.638,6.571c0.098,0.025,0.225,0.061,0.365,0.117-0.117-0.029-0.242-0.061-0.371-0.092l-2.296,9.205c-0.174,0.432-0.615,1.08-1.609,0.834,0.035,0.051-2.552-0.637-2.552-0.637l-1.743,4.019,4.569,1.139c0.85,0.213,1.683,0.436,2.503,0.646l-1.453,5.834,3.507,0.875,1.439-5.772c0.958,0.26,1.888,0.5,2.798,0.726l-1.434,5.745,3.511,0.875,1.453-5.823c5.987,1.133,10.489,0.676,12.384-4.739,1.527-4.36-0.076-6.875-3.226-8.515,2.294-0.529,4.022-2.038,4.483-5.155zm-8.022,11.249c-1.085,4.36-8.426,2.003-10.806,1.412l1.928-7.729c2.38,0.594,10.012,1.77,8.878,6.317zm1.086-11.312c-0.99,3.966-7.1,1.951-9.082,1.457l1.748-7.01c1.982,0.494,8.365,1.416,7.334,5.553z"
                      />
                    </g>
                  </svg>
                  0.00069079
                </div>
              </div>
              <div
                class="section__row"
              >
                <div
                  class="section__label"
                >
                  BONUS AMOUNT
                </div>
                <div
                  class="section__value section__value__amount section__value__bonus"
                >
                  <svg
                    class="fill"
                    viewBox="0 0 64 64"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      transform="translate(0.00630876,-0.00301984)"
                    >
                      <path
                        d="m63.033,39.744c-4.274,17.143-21.637,27.576-38.782,23.301-17.138-4.274-27.571-21.638-23.295-38.78,4.272-17.145,21.635-27.579,38.775-23.305,17.144,4.274,27.576,21.64,23.302,38.784z"
                      />
                      <path
                        class="innerFill"
                        d="m46.103,27.444c0.637-4.258-2.605-6.547-7.038-8.074l1.438-5.768-3.511-0.875-1.4,5.616c-0.923-0.23-1.871-0.447-2.813-0.662l1.41-5.653-3.509-0.875-1.439,5.766c-0.764-0.174-1.514-0.346-2.242-0.527l0.004-0.018-4.842-1.209-0.934,3.75s2.605,0.597,2.55,0.634c1.422,0.355,1.679,1.296,1.636,2.042l-1.638,6.571c0.098,0.025,0.225,0.061,0.365,0.117-0.117-0.029-0.242-0.061-0.371-0.092l-2.296,9.205c-0.174,0.432-0.615,1.08-1.609,0.834,0.035,0.051-2.552-0.637-2.552-0.637l-1.743,4.019,4.569,1.139c0.85,0.213,1.683,0.436,2.503,0.646l-1.453,5.834,3.507,0.875,1.439-5.772c0.958,0.26,1.888,0.5,2.798,0.726l-1.434,5.745,3.511,0.875,1.453-5.823c5.987,1.133,10.489,0.676,12.384-4.739,1.527-4.36-0.076-6.875-3.226-8.515,2.294-0.529,4.022-2.038,4.483-5.155zm-8.022,11.249c-1.085,4.36-8.426,2.003-10.806,1.412l1.928-7.729c2.38,0.594,10.012,1.77,8.878,6.317zm1.086-11.312c-0.99,3.966-7.1,1.951-9.082,1.457l1.748-7.01c1.982,0.494,8.365,1.416,7.334,5.553z"
                      />
                    </g>
                  </svg>
                  +
                  0.00001234
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      style="height: 768px;"
    >
      <div
        aria-modal="true"
        class="circle_warning modal scale--enter scale--enter-active"
        role="dialog"
      >
        <div
          class="modal__header"
        >
          <div
            class="modal__header__headline"
          >
            <svg
              class="icon icon__warning"
              viewBox="0 0 35 35"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                data-name="Layer 2"
                id="Layer_2"
              >
                <g
                  data-name="Слой 5"
                  id="Слой_5"
                >
                  <path
                    class="cls-1"
                    d="M17.5,0A17.5,17.5,0,1,0,35,17.5,17.51,17.51,0,0,0,17.5,0Zm2.56,24.54a2.14,2.14,0,0,1-2.14,2.14h-.83A2.14,2.14,0,0,1,15,24.54V17.69a2.14,2.14,0,0,1,2.14-2.15h.83a2.14,2.14,0,0,1,2.14,2.15ZM17.5,13.3a3.09,3.09,0,1,1,3.1-3.09A3.09,3.09,0,0,1,17.5,13.3Z"
                  />
                </g>
              </g>
            </svg>
            BONUS DETAILS
          </div>
          <div
            class="modal__header__close"
          >
            <svg
              class="icon icon__close"
              viewBox="0 0 365.696 365.696"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M243.188 182.86L356.32 69.726c12.5-12.5 12.5-32.766 0-45.247L341.238 9.398c-12.504-12.503-32.77-12.503-45.25 0L182.86 122.528 69.727 9.374c-12.5-12.5-32.766-12.5-45.247 0L9.375 24.457c-12.5 12.504-12.5 32.77 0 45.25l113.152 113.152L9.398 295.99c-12.503 12.503-12.503 32.769 0 45.25L24.48 356.32c12.5 12.5 32.766 12.5 45.247 0l113.132-113.132L295.99 356.32c12.503 12.5 32.769 12.5 45.25 0l15.081-15.082c12.5-12.504 12.5-32.77 0-45.25zm0 0"
              />
            </svg>
          </div>
        </div>
        <div
          class="modal__main"
        >
          <div
            class="section"
          >
            <div
              class="section__row"
            >
              <div
                class="section__label"
              >
                Date / time
              </div>
              <div
                class="section__value"
              >
                02/16/2021 13:44:11
              </div>
            </div>
            <div
              class="section__row"
            >
              <div
                class="section__label"
              >
                CLAIM AT
              </div>
              <div
                class="section__value"
              >
                02/18/2021 01:41:36
              </div>
            </div>
            <div
              class="section__row"
            >
              <div
                class="section__label"
              >
                TYPE
              </div>
              <div
                class="section__value"
              >
                DAILY
              </div>
            </div>
            <div
              class="section__row"
            >
              <div
                class="section__label"
              >
                WAGER
              </div>
              <div
                class="section__value section__value__amount"
              >
                <svg
                  class="fill"
                  viewBox="0 0 64 64"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    transform="translate(0.00630876,-0.00301984)"
                  >
                    <path
                      d="m63.033,39.744c-4.274,17.143-21.637,27.576-38.782,23.301-17.138-4.274-27.571-21.638-23.295-38.78,4.272-17.145,21.635-27.579,38.775-23.305,17.144,4.274,27.576,21.64,23.302,38.784z"
                    />
                    <path
                      class="innerFill"
                      d="m46.103,27.444c0.637-4.258-2.605-6.547-7.038-8.074l1.438-5.768-3.511-0.875-1.4,5.616c-0.923-0.23-1.871-0.447-2.813-0.662l1.41-5.653-3.509-0.875-1.439,5.766c-0.764-0.174-1.514-0.346-2.242-0.527l0.004-0.018-4.842-1.209-0.934,3.75s2.605,0.597,2.55,0.634c1.422,0.355,1.679,1.296,1.636,2.042l-1.638,6.571c0.098,0.025,0.225,0.061,0.365,0.117-0.117-0.029-0.242-0.061-0.371-0.092l-2.296,9.205c-0.174,0.432-0.615,1.08-1.609,0.834,0.035,0.051-2.552-0.637-2.552-0.637l-1.743,4.019,4.569,1.139c0.85,0.213,1.683,0.436,2.503,0.646l-1.453,5.834,3.507,0.875,1.439-5.772c0.958,0.26,1.888,0.5,2.798,0.726l-1.434,5.745,3.511,0.875,1.453-5.823c5.987,1.133,10.489,0.676,12.384-4.739,1.527-4.36-0.076-6.875-3.226-8.515,2.294-0.529,4.022-2.038,4.483-5.155zm-8.022,11.249c-1.085,4.36-8.426,2.003-10.806,1.412l1.928-7.729c2.38,0.594,10.012,1.77,8.878,6.317zm1.086-11.312c-0.99,3.966-7.1,1.951-9.082,1.457l1.748-7.01c1.982,0.494,8.365,1.416,7.334,5.553z"
                    />
                  </g>
                </svg>
                0.00069079
              </div>
            </div>
            <div
              class="section__row"
            >
              <div
                class="section__label"
              >
                BONUS AMOUNT
              </div>
              <div
                class="section__value section__value__amount section__value__bonus"
              >
                <svg
                  class="fill"
                  viewBox="0 0 64 64"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    transform="translate(0.00630876,-0.00301984)"
                  >
                    <path
                      d="m63.033,39.744c-4.274,17.143-21.637,27.576-38.782,23.301-17.138-4.274-27.571-21.638-23.295-38.78,4.272-17.145,21.635-27.579,38.775-23.305,17.144,4.274,27.576,21.64,23.302,38.784z"
                    />
                    <path
                      class="innerFill"
                      d="m46.103,27.444c0.637-4.258-2.605-6.547-7.038-8.074l1.438-5.768-3.511-0.875-1.4,5.616c-0.923-0.23-1.871-0.447-2.813-0.662l1.41-5.653-3.509-0.875-1.439,5.766c-0.764-0.174-1.514-0.346-2.242-0.527l0.004-0.018-4.842-1.209-0.934,3.75s2.605,0.597,2.55,0.634c1.422,0.355,1.679,1.296,1.636,2.042l-1.638,6.571c0.098,0.025,0.225,0.061,0.365,0.117-0.117-0.029-0.242-0.061-0.371-0.092l-2.296,9.205c-0.174,0.432-0.615,1.08-1.609,0.834,0.035,0.051-2.552-0.637-2.552-0.637l-1.743,4.019,4.569,1.139c0.85,0.213,1.683,0.436,2.503,0.646l-1.453,5.834,3.507,0.875,1.439-5.772c0.958,0.26,1.888,0.5,2.798,0.726l-1.434,5.745,3.511,0.875,1.453-5.823c5.987,1.133,10.489,0.676,12.384-4.739,1.527-4.36-0.076-6.875-3.226-8.515,2.294-0.529,4.022-2.038,4.483-5.155zm-8.022,11.249c-1.085,4.36-8.426,2.003-10.806,1.412l1.928-7.729c2.38,0.594,10.012,1.77,8.878,6.317zm1.086-11.312c-0.99,3.966-7.1,1.951-9.082,1.457l1.748-7.01c1.982,0.494,8.365,1.416,7.334,5.553z"
                    />
                  </g>
                </svg>
                +
                0.00001234
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
